### PR TITLE
Update examples securing Verifiable Presentations

### LINKED ISSUE #222: Potential problem with Example 2
There may be a problem with [Example 2](https://www.w3.org/TR/vc-jose-cose/#example-a-simple-example-of-a-verifiable-presentation), similar to the one reported in #217.

TL;DR: the block on `VerifiableCredential` may have to be exchanged to something like this:

```json
        {[
		 "@context": ["https://www.w3.org/ns/credentials/v2"], 
		 "type": ["EnvelopedVerifiableCredential"], 
		 "id": "data:application/vcld+json+sd-jwt;eyJhbGciOiJFUzM4NCIsImtpZ…" 
	 }] 
```

The reason: the [definition of the `verifiableCredential` property](https://www.w3.org/TR/vc-data-model-2.0/#defn-verifiableCredential) requires the value to be a `VerifiableCredential` or an `EnvelopedVerifiableCredential`. The question that comes up is whether a `VerifiableCredential` object ***without a securing mechanism*** (a.k.a. proof) is allowed within a verifiable presentation or not. The claim is that the answer is 'no' (@msporny, you should chime in on this), although, I must admit, this is not crystal clear in the VCDM spec, and it should, maybe, be made clearer.

If we agree that this is the case, then the question is how that would work in combination with JOSE: can an enveloping proof like JOSE be encapsulated in another JOSE payload? Because if the answer is no, then the only viable approach is to use the `EnvelopedVerifiableCredential` approach above. (Unless we would use DI for the securing of the credential, which would be a mix of two securing paradigms; probably a no-no.)

Another possibility is that we make it less stringent in the VCDM, and we do allow a credential in that position, i.e., without a securing mechanism.

Note also the related [Figure 10](https://www.w3.org/TR/vc-data-model-2.0/#info-graph-vp-jwt) in the VCDM.

(I marked this as 'before CR' because it does raise some fundamental issues about JOSE-COSE/VCDM relationship.)

CC @decentralgabe @msporny @dlongley

### PR DESCRIPTION
**Pull request by decentralgabe**
Fix #222 

* Update example to use the enveloped type

For both JOSE and COSE
* Add normative guidance around presentations using the enveloped type
* Add normative guidance around securing only secured credentials

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/pull/225.html" title="Last updated on Jan 30, 2024, 5:52 PM UTC (6ab617b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/225/3c476fc...6ab617b.html" title="Last updated on Jan 30, 2024, 5:52 PM UTC (6ab617b)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -300,21 +308,12 @@ <h2>Securing JSON-LD Verifiable Presentations with JOSE</h2>
   ],
   "type": "VerifiablePresentation",
   "verifiableCredential": [{
-    "@context": [
-      "https://www.w3.org/ns/credentials/v2",
-      "https://www.w3.org/ns/credentials/examples/v2"
-    ],
+    "@context": ["https://www.w3.org/ns/credentials/v2"],
     "id": "http://university.example/credentials/1872",
```
Comment (davidlehn) [inline index.html:312]: Two `id` fields needs to be fixed.

**Review by iherman — APPROVED** [review]

**Review by TallTed — APPROVED** [review]

**Review by brentzundel — APPROVED** [review]

**Comment by David-Chadwick** [comment]
This change adds the media type of sd-jwt for the enveloped VC, but currently there is no sd-jwt VC example in the draft. It would be better if the enveloped VC was pure JSON protected and the media type reflected this.

**Comment by decentralgabe** [comment]
@David-Chadwick merging this for agility. please open/add another issue if you feel this is insufficient.

**Comment by David-Chadwick** [comment]
I have added this issue as a comment to #204

**Pull request merged**